core/primitives/tipset/tipset.cpp

```diff
@@ -57,12 +57,10 @@ namespace fc::primitives::tipset {
   using vm::message::SignedMessage;
   using vm::message::UnsignedMessage;
 
-  MessageVisitor::~MessageVisitor() {
-    if (state_tree) {
-      delete state_tree;
-      state_tree = nullptr;
-    }
-  }
+  MessageVisitor::~MessageVisitor() = default;
```

Comment:
Could we move it to .hpp?

Reply:
No, because of unique_ptr to forward-declared type